cmd/gitserver/server/vcs_dependencies_syncer.go

```
@@ -108,6 +137,14 @@ func (s *vcsPackagesSyncer) Fetch(ctx context.Context, remoteURL *vcs.URL, dir G
 		return err
 	}
 
+	if revspec != "" {
+		requestedVersion := strings.TrimSuffix(strings.TrimPrefix(revspec, "v"), "^0")
+		err = s.lazySyncRequestedVersion(ctx, dir, depName, versions, requestedVersion)
+		if err == nil {
```

What happens if there is an error?

---

It's silently ignored. There's a new test case to demonstrate this scenario https://github.com/sourcegraph/sourcegraph/pull/38811/files#diff-08a8446c4af08a069239227b1cc36ca8b2ff546ded694038d4322b8b39e6a8dfR179

---

Added a comment why we silently ignore errors here 👍